erpnext/hr/doctype/full_and_final_statement/full_and_final_statement.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.utils import get_link_to_form, today, flt
+from frappe.model.document import Document
+
+class FullAndFinalStatement(Document):
+
+	def validate(self):
+		self.get_outstanding_statements()
+		if self.docstatus == 1:
+			self.validate_settlement('payables')
+			self.validate_settlement('receivables')
+
+	def validate_settlement(self, type):
+		for data in self.get(type,[]):
+			if data.status == "Unsettled":
+				frappe.throw(_("Settled all Payables and Receivable before submission"))
+
+
+	@frappe.whitelist()
+	def get_outstanding_statements(self):
+		if self.relieving_date:
+			if not len(self.get("payables", [])):
+				components = self.get_payable_component()
+				self.create_component_row(components, "payables")
+			if not len(self.get("receivables", [])):
+				components = self.get_receivable_component()
+				self.create_component_row(components, "receivables")
+
+			if not len(self.get("assets_allocated", [])):
+				for data in self.get_assets_movement():
+					self.append("assets_allocated", data)
+		else:
+			frappe.throw(_("Set Relieving Date for Employee: {0}").format(get_link_to_form("Employee", self.employee)))
+
+	def create_component_row(self, components, type):
+		data = {}
+		for component in components:
+			data["status"] = "Unsettled"
+			data["component"] = component
+			self.append(type, data)
+
+
+	def get_payable_component(self):
+		return [
+			"Salary Slip",
+			"Gratuity",
+			"Expense Claim",
+			"Bonus",
+			"Leave Encashment",
```

Comment:
No, we cant, The component is only the name given It can be also like an Outstanding salary slip or advanced Salary slip It can be settled in other ways too.

Reply:
We can still set the reference doctype and users can change it if they want to, as in most cases the doctypes would be standard:

Salary Slip
Gratuity
Expense Claim
Additional Salary
Leave Encashment

Reply:
Done